docs/aws/audit/cloudwatchmonitoring/rules/cloudwatch_log_groups_not_encrypted.mdx

```diff
@@ -22,98 +22,89 @@ HIPAA,PCIDSS,GDPR,CISAWS,CBP,NIST,SOC2,AWSWAF,SEBI,RBI_UCB
-
-3. Verify that the CloudWatch Log Groups are now encrypted:
-```bash
-aws logs describe-log-groups --query 'logGroups[?kmsKeyId!=null].logGroupName' --output text
-```
-
-By following these steps, you can remediate the misconfiguration of CloudWatch Log Groups not being encrypted in AWS using AWS CLI.
-
-#### Using Python
-
-To remediate the misconfiguration of CloudWatch Log Groups not being encrypted in AWS using Python, you can follow these steps:
-
-1. Install the AWS SDK for Python (Boto3) if you haven't already:
-```bash
-pip install boto3
-```
-
-2. Use the following Python script to enable encryption for all existing CloudWatch Log Groups:
-
-```python
-import boto3
-
-def encrypt_cloudwatch_logs():
-    # Initialize the CloudWatch Logs client
-    client = boto3.client('logs')
-
-    # Get a list of all existing CloudWatch Log Groups
-    log_groups = client.describe_log_groups()
-
-    # Enable encryption for each log group
-    for log_group in log_groups['logGroups']:
-        log_group_name = log_group['logGroupName']
-        
-        # Enable encryption for the log group
-        response = client.put_retention_policy(
-            logGroupName=log_group_name,
-            retentionInDays=7,
-            metricFilterName='AllLogs',
-            metricTransformations=[
-                {
-                    'metricName': 'LogEvents',
-                    'metricNamespace': 'CloudWatchLogs',
-                    'metricValue': '1'
-                }
-            ]
-        )
-
-        print(f"Enabled encryption for Log Group: {log_group_name}")
-
-if __name__ == '__main__':
-    encrypt_cloudwatch_logs()
-```
-
-3. Run the Python script, and it will enable encryption for all existing CloudWatch Log Groups in your AWS account.
-
-Please note that this script assumes you have the necessary permissions to modify CloudWatch Log Groups in your AWS account. Make sure to configure your AWS credentials properly before running the script.
-
-### Additional Reading:
\ No newline at end of file
+### Using Console:
+
+1. **Create an AWS KMS Key:**
+   - Go to the AWS KMS console.
+   - Click on "Create key" and follow the prompts to create a symmetric key.
+   - Make note of the Key ID and ARN provided.
+
+2. **Set Permissions on the KMS Key:**
+   - After creating the key, navigate to it in the AWS KMS console.
+   - Select "Key policy" from the left-hand menu.
+   - Edit the key policy to allow CloudWatch Logs service principal to use the key.
+   - Add permissions as described in the documentation under "Step 2: Set permissions on the KMS key".
+
+3. **Associate the KMS Key with a Log Group:**
+   - Navigate to the CloudWatch Logs console.
+   - Find the log group you want to encrypt or create a new log group.
+   - During log group creation or by selecting an existing log group, associate the KMS key with it.
```

Comment:
Log group encryption can not be changed from console. See here - https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/encrypt-log-data-kms.html#encryption-limits

Reply:
reverted